src/Ocelot/DependencyInjection/ConfigurationBuilderExtensions.cs

```diff
@@ -28,37 +29,42 @@ public static IConfigurationBuilder AddOcelotBaseUrl(this IConfigurationBuilder
             return builder;
         }
 
-        public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder)
+        public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder, IHostingEnvironment env = null)
```

Comment:
I agree... optional arguments usually the fastest, but not the best way to solve the problem. Plus the API looks a bit messy in that case. I'd prefer overloaded versions of the method.
But in this case (as far as I understand) you think it would be better to always provide IHostingEnvironment instance (even if user may never use env specific configuration files).
Anyway, I'll make env a required argument in next commit.